### PR TITLE
feat: create-rspack add template-vue

### LINKED ISSUE #2107: [Feature Request]: add template-vue for create-rspack
### What problem does this feature solve?

Currently `create-rspack` only has react template, can't create other project with rspack

### What does the proposed API of configuration look like?

add prompt for users to select their preferred framework, and I'm willing work for it.

### PR DESCRIPTION
**Pull request by faga295**
## Summary
closed #2107 
It has some differences from `examples/vue`:
* remove less loader
* remove `builtins/define` and `devServer` in `rspack.config.js`(I'm not entirely sure are these necessary)

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run changeset`.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by hardfist** [comment]
hold until we supports vue-loader

File: packages/create-rspack/template-vue/src/components/HelloWorld.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<script >
```
Comment (fwqaaq) [inline packages/create-rspack/template-vue/src/components/HelloWorld.vue:1]:
```suggestion
<script>
```

**Comment by faga295** [comment]
@hardfist  cc

File: packages/create-rspack/index.js [hunk 2]
```diff
@@ -45,7 +45,8 @@ yargs(hideBin(process.argv))
 				message: "Project template",
 				choices: [
 					{ title: "react", value: "react" },
-					{ title: "react-ts", value: "react-ts" }
+					{ title: "react-ts", value: "react-ts" },
+					{ title: "vue", value: "vue" }
```
Comment (hardfist) [inline packages/create-rspack/index.js:49]: maybe we should use "vue3" here

Comment (faga295) [reply packages/create-rspack/index.js:49]: Currently vue3 is the default version of vue, I don't think we need to point out the vue version, vue2 is the one that needs to point out

Comment (hardfist) [reply packages/create-rspack/index.js:49]: I'm not a vue user, it that's the case then it's ok for me

**Review by hardfist — APPROVED** [review]

**Pull request merged**